python/dgl/merge.py

```diff
@@ -0,0 +1,103 @@
+"""Utilities for merging graphs."""
+
+import dgl
+from . import backend as F
+from .base import DGLError
+
+__all__ = ['merge']
+
+def merge(graphs):
+    r"""Merge a sequence of graphs together into a single graph.
+
+    Nodes and edges that exist in ``graphs[i+1]`` but not in ``dgl.merge(graphs[0:i+1])``
+    will be added to ``dgl.merge(graphs[0:i+1])`` along with their data.
```

Comment:
When will we merge graphs with different number of nodes?  I think restricting them to have the same number of nodes will make things easier, and will also match the semantics of graph union (i.e. two graphs having the same nodes has the edge sets unioned).

Reply:
The feature request that prompted this PR specifically gave an example of merging graphs with different numbers of nodes, so I think it's appropriate for the operation to accommodate that use case.